File: src/main/java/pl/tsimur/jpatest/model/Subscription.java
package pl.tsimur.jpatest.model;


import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

import static javax.persistence.EnumType.STRING;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Table
@Entity
public class Subscription {
    @Id
    @Enumerated(STRING)
    private Tariff name;

    private Float price;

    private String title;

    @OneToMany(mappedBy = "subscription", fetch = FetchType.EAGER)
    @JsonIgnore
    private List<User> users = new ArrayList<>();

}

File: src/main/kotlin/pl/timur/jpatest/model/Subscription.kt
package pl.timur.jpatest.model

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.*

@Table
@Entity
data class Subscription(
    @Id
    @Enumerated(EnumType.STRING)
    val name: Tariff,
    val price: Float,
    val title: String
) {
    @OneToMany(mappedBy = "subscription", fetch = FetchType.EAGER)
    @JsonIgnore
    val users: List<User> = ArrayList()
}
